RecoBTag/DeepFlavour/plugins/DeepFlavourJetTagProducer.cc

```diff
@@ -0,0 +1,261 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/DeepFormats/interface/DeepFlavourTagInfo.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "RecoBTag/DeepFlavour/interface/tensor_fillers.h"
+
+// Declaration of the data structure that is hold by the edm::GlobalCache.
+// In TensorFlow, the computational graph is stored in a stateless meta graph object which can be
+// shared by multiple session instances which handle the initialization of variables related to the
+// meta graph. Following this approach in CMSSW, a meta graph should be stored in a GlobalCache
+// which can be accesses by sessions owned by multiple stream module copies. Instead of using only
+// the plain meta graph, we make use of a Cache struct that can be extended in the future if nedded.
+// In addition, the meta graph is protected via std::atomic, which should not affect the performance
+// as it is only accessed in the module constructor and not in the actual produce loop.
+struct Cache {
+  Cache() : metaGraph(nullptr)
```

Comment:
is there a possibility for symbol collision here, or is this visible only locally?

Reply:
Maybe we should rename it to `DeepFlavourJetTagProducerCache` to ensure it's unique.

Reply:
Or put it in anonymous namespace?